AnkiDroid/src/main/java/com/ichi2/anki/instantnoteeditor/InstantEditorViewModel.kt

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (c) 2024 Ashish Yadav <mailtoashish693@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation; either version 3 of the License, or (at your option) any later
+ * version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.ichi2.anki.instantnoteeditor
+
+import android.app.Activity
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.ichi2.anki.CollectionManager.withCol
+import com.ichi2.anki.NoteFieldsCheckResult
+import com.ichi2.anki.OnErrorListener
+import com.ichi2.anki.R
+import com.ichi2.anki.checkNoteFieldsResponse
+import com.ichi2.anki.withProgress
+import com.ichi2.libanki.Collection
+import com.ichi2.libanki.DeckId
+import com.ichi2.libanki.Decks
+import com.ichi2.libanki.Note
+import com.ichi2.libanki.NotetypeJson
+import com.ichi2.libanki.undoableOp
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.launch
+import timber.log.Timber
+
+class InstantEditorViewModel : ViewModel(), OnErrorListener {
+    override val onError = MutableSharedFlow<String>()
+
+    val clozeNoteType = MutableSharedFlow<Boolean>()
+
+    val instantEditorError = MutableSharedFlow<String?>()
+
+    val noteSaved = MutableSharedFlow<Boolean>()
+
+    // TODO: When the note is saved the previous notetype should be
+    //  selected back, right now it doesn't work as expected
+    val oldNoteId = MutableSharedFlow<Long>()
+
+    private val _editorNote = MutableLiveData<Note?>()
+    val editorNote: LiveData<Note?> get() = _editorNote
+
+    private val _currentlySelectedNotetype = MutableLiveData<NotetypeJson>()
+    val currentlySelectedNotetype: LiveData<NotetypeJson> get() = _currentlySelectedNotetype
+
+    private val _deckId = MutableLiveData<DeckId>(0)
+    val deckId: LiveData<DeckId> get() = _deckId
+
+    private val _savingNote = MutableLiveData(false)
+    val savingNote: LiveData<Boolean> get() = _savingNote
+
+    init {
+        viewModelScope.launch {
+            oldNoteId.emit(-1)
+        }
+    }
+
+    fun setDeckId(deckId: DeckId) {
+        _deckId.value = deckId
+    }
+
+    fun setSavingNote(value: Boolean) {
+        _savingNote.value = value
+    }
+
+    suspend fun getCurrentDeck() {
+        withCol {
+            _deckId.postValue(config.get(Decks.CURRENT_DECK) ?: 1L)
+        }
```

Comment:
Don't do things which don't need the collection inside `withCol`

Get the value from the collection, then unlock the collection

```suggestion
        val did = withCol { config.get(Decks.CURRENT_DECK) ?: 1L }
        _deckId.postValue(did)
```